hibernate-core/src/main/java/org/hibernate/action/internal/EntityInsertAction.java

```diff
@@ -150,6 +153,168 @@ public void execute() throws HibernateException {
 
 		handleNaturalIdPostSaveNotifications( id );
 
+		EntityMetamodel entityMetamodel = getPersister().getEntityMetamodel();
+
+		if (entityMetamodel.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading()) {
+			PersistentAttributeInterceptable interceptable = ((PersistentAttributeInterceptable) instance );
+			PersistentAttributeInterceptor interceptor = interceptable.$$_hibernate_getInterceptor();
+			final BytecodeEnhancementMetadata enhancementMetadata = entityMetamodel.getBytecodeEnhancementMetadata();
+			final LazyAttributesMetadata lazyAttributesMetadata = enhancementMetadata.getLazyAttributesMetadata();
+			if (interceptor == null) {
+				Type[] types = persister.getPropertyTypes();
+				final String[] propertyNames = persister.getPropertyNames();
+				final CascadeStyle[] cascadeStyles = persister.getPropertyCascadeStyles();
+				boolean needsInterceptor = false;
+				Set<String> initializedLazyFields = new HashSet<>();
+				for ( int i = 0; i < types.length; i++) {
+					if(!types[i].isAssociationType()) {
+						continue;
+					}
```

Comment:
Yes, that is essential.  Just having a value doesn't mean Hibernate Search can traverse any association properties  during indexing unless interceptors are installed.  Assume the value was set by the client, and is "dead" from a perspective of indexing.  Imagine that the present object is only a small part of a huge multidimensional object graph, and that your search indexes depend on being able to traverse both up and down the hierarchical tree, using a proxy installed at every *ToOne and *ToMany property, otherwise your index will be incomplete and corrupt!